recipes/quantlib/all/conanfile.py

```diff
@@ -43,27 +43,38 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.80.0")
+        self.requires("boost/1.80.0", transitive_headers=True)
 
     def validate(self):
         if self.info.settings.compiler.get_safe("cppstd"):
-            check_min_cppstd(self, 11)
+            check_min_cppstd(self, 14 if self.version >= "1.24" else 11)
         if self.info.settings.compiler == "gcc" and Version(self.info.settings.compiler.version) < "5":
             raise ConanInvalidConfiguration("gcc < 5 not supported")
+        if self.version >= "1.24" and is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration("MSVC DLL build is not supported by upstream")
```

Comment:
I don't understand, is CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS not sufficient? Why is it not injected for this specific version?

Reply:
I don't know why upstream does not allow shared build with MSVC, and I don't have a Windows device to try, so I just leave it consistent with upstream instead of a patch.

Reply:
It makes recipe inconsistent. I don't understand your answer. Is it prevented or is it broken by default without `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` like previous version.

Reply:
It's explicitly prevented in the upstream CMake script.

Reply:
It's patched in 1.22, I don't see any reason to not do the same in 1.30 since it works just fine and it's a one line patch.

Reply:
You can add it back in a subsequent pull request. I don't have a Windows device to test if it works so I'd better be conservative in this pull request.